### ISSUE TITLE
Unresponsive and black/blank screen  on the new UI

### ISSUE THREAD

**Issue by deanrih** [ValveSoftware/steam-for-linux#9386]
#### Your system information

* Steam client version (build number or date): 1682634349
* Distribution (e.g. Ubuntu): Manjaro (Arch)
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:

I just resubmitting the bug from the Steam Community [here](https://steamcommunity.com/groups/SteamClientBeta/discussions/0/3826413850813915465/)

After the new Beta update, some parts (mostly that behave like a browser, like stores, community, profile, etc.) becomes unresponsive when first opened (especially on the new Steam browser) and can be resolved by switching screen/tab (e.g from Store to Library and back to Store, or switching from Tab 1 to Tab 2 back to Tab 1 on the Steam Browser) or resize the window, and some parts becomes blank when the window is maximized and restored (but not on manually resized) and have to switch screen/tab (e.g from Store to Library and back to Store, or switching from Tab 1 to Tab 2 back to Tab 1 on the Steam Browser)

### Unresponsiveness

1. Open the Steam Client
2. Open a community discussion
3. Go to the Steam Client Beta community
4. Middle click on any discussion (so it will open on the new Steam Browser window)
5. The page is can't be interacted with and can't be scrolled until resized
6. Middle click on any discussion or url from this Steam Browser
7. Go to the newly created tab
8. The page is can't be interacted with and can't be scrolled until resized or back to the previous tab and back again to the newly created tab

### Blank part the UI

* This is happened both on the main window of Steam Client UI and the Steam Browser
* The blankness can't be resolved by refreshing the page, it can only be resolved by switching tab on the Steam Browser or switching to the Library view first on the main Steam Client UI
* On the main Steam Client UI, it seems the Maximize and Restore window button doesn't have any issue which means doesn't trigger this bug.

1. Open the Steam Client
2. Open a community discussion
3. Maximize the window by dragging the window to the top (Clicking the maximize button seems doesn't have any issue)
4. The view is now only show empty black rectangle
5. Go to the library
6. Go back to the community discussion
7. The view is now normal again
8. Restore the window by dragging the window from the top (Clicking the maximize/restore button seems doesn't have any issue)
9. The view is now only show empty black rectangle
10. Go to the library
11. Go back to the community discussion
12. The view is now normal again

**Comment by A-UNDERSCORE-D** [comment]
Confirming for both; OpenSUSE Tumbleweed

**Comment by TheTee82** [comment]
Same here on:
- Arch
  - Graphics Platform: X11
  - KDE Plasma: 5.27.4
  - KDE Frameworks: 5.105.0
  - Qt Version: 5.15.9

Only happens when manually maximizing (by dragging the window to top) and restoring from maximized to normal (dragging window back).
Weirdly enough, when dragging the steam client window to the left or right side (snapping to side and autosize the window to be half the screen) the client works perfectly fine. Only happens when maximizing (manually dragging) and restoring to original window-size (manually dragging).

**Comment by dsalt** [comment]
Re. unresponsive UI – I'm fairly sure that there's a synchronisation bug here. On first opening a web page, the page area is unresponsive; but resize it, and either the original area or the whole area becomes responsive.

How much becomes responsive? If you use the size toggle, only the original (previously-displayed) area; if you drag-resize, the whole area.

(Maximising then starting and cancelling a drag-resize also restores responsiveness.)

I've not seen any blank areas in the UI, with the exception of the “blank browser” bug. (AMD/Mesa.)

**Comment by banchaa** [comment]
Same for me, additionally, the small mode of the beta client always has a black window inside the content.

OS: Debian GNU/Linux 12 (bookworm) x86_64 
Kernel: 6.1.0-7-amd64 
DE: Plasma 5.27.2
WM: KWin

**Comment by DanMan** [comment]
UI is unresponsive for me as well after launching Steam. Only after resizing the window I can interact with stuff and scroll with the mouse wheel. It never goes black for me though.

Fedora 36, Nvidia GPU, X.org

**Comment by appleneko2001** [comment]
I have same issues with my Steam beta. The content part just goes black after I resized my Steam window. Looks like only main window affected. Workaround is switch to other tab. For example, click to LIBRARY tab from STORE when the content part goes black in store tab, then switch back to STORE and let it reload.

Version: 1683859267
OS: Debian GNU/Linux 12 (bookworm) x86_64
Kernel: xanmod 6.1.14-x64v3
DE: KDE Plasma 5.27.2
WM: KWin
Graphics: GeForce RTX 3060 with proprietary driver 510.108.03 (cant use latest driver since its conflicting to my monitor lol)
How I installed my Steam: official deb package (no any sandboxing)

**Comment by dsalt** [comment]
@appleneko2001, it looks like you're describing #9484.

**Comment by DanMan** [comment]
The recent beta client with HW accel. enabled for Nvidia GPUs has fixed the "non-interactive shop until window is resized" problem as well for me.

The navigation is still glitching though. It's still a matter of chance if the dropdowns show up on mouse hover or not. But I guess that's a different issue.

**Comment by tuxiqae** [comment]
Not sure whether it's related but CS:GO completely freezes when I open in-game overlay when the new client is enabled

**Comment by Fesiug** [comment]
I have this issue too, with the same workaround. Up-to-date Arch Linux, KDE, X11, Nvidia GPU

**Comment by rKsanu2MMYvypWePtQWM** [comment]
Same for the black screen issue.

**Comment by deanrih** [comment]
I don't know since what version it is solved, but my current version is 1686453716.

It seems that the unresponsive problem is solved, at least for me or at least I haven't encountered one, yet. But the blank screen is still happened when you try to maximize and restore the window by dragging the window up to the top edge of the screen or from the top edge of the screen respectively, again, clicking the Maximize and Restore button seems doesn't have this problem.

**Comment by rKsanu2MMYvypWePtQWM** [comment]
Black screen is definitely still happening. I was restoring steam thru the system tray and it's black except for library. After visiting library it gets fixed.

**Comment by DanielHuelsmann** [comment]
I have the same bug. Even when maximizing and unmaximizing the Steam client window.

Steam client version: 1687386907 (via Flatpak)
Operating System: Fedora Linux 38
KDE Plasma Version: 5.27.5
Kernel Version: 6.3.8-200.fc38.x86_64 (64-bit)
Graphics Platform: Wayland
Processors: 12 × AMD Ryzen 5 5600X 6-Core Processor
Memory: 15.5 GiB of RAM
Graphics Processor: AMD Radeon RX 6600

**Comment by MerkaST** [comment]
I can't reproduce the black screen on the maximise/restore action because my window manager doesn't use that gesture, but for me it occurs when I minimise and then restore Steam.
Lubuntu 22.04
LXQt/Openbox
AMD R9 380X

**Comment by TheTee82** [comment]
Btw. black window bug is also happening on Steam Deck in Desktop Mode, when trying "drag to border maximizing" nad having the Store page open.

SteamOS: 3.4.9
Steam client: 1687386907

**Comment by Plaoo** [comment]
Same problem, my hardware and os:

Steam Version:  1687386907
OS: openSUSE Tumbleweed x86_64
Host: GF65 Thin 9SD REV:1.0
Kernel: 6.3.9-1-default
Shell: zsh 5.9
Resolution: 2560x1080
DE: Plasma 5.27.6
WM: KWin
Terminal: tmux
CPU: Intel i5-9300H (8) @ 4.100GHz
GPU: Intel CoffeeLake-H GT2 [UHD Graphics 630]
GPU: NVIDIA GeForce GTX 1660 Ti Mobile
Memory: 32 Gb

**Comment by Caffeineix** [comment]
This is still an issue on the latest beta.

Video Card:
Driver: NVIDIA Corporation NVIDIA GeForce RTX 3060/PCIe/SSE2
Driver Version: 4.6.0 NVIDIA 525.78.01
OpenGL Version: 4.6
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 60 Hz
VendorID: 0x10de
DeviceID: 0x2504
Revision Not Detected
Number of Monitors: 1
Number of Logical Video Cards: 1
Primary Display Resolution: 3840 x 2160
Desktop Resolution: 3840 x 2160
Primary Display Size: 24.88" x 14.17" (28.62" diag), 63.2cm x 36.0cm (72.7cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 12288 MB
Supported MSAA Modes: 2x 4x 8x 16x

**Comment by faithvoid** [comment]
Can confirm the same issue on my system.

OS: Arch Linux x86_64 
Host: Inspiron 620 00 
Kernel: 6.3.9-zen1-1-zen 
Resolution: 2560x1440 
DE: Xfce 4.18 
CPU: Intel i7-2600 (8) @ 3.400GHz 
GPU: NVIDIA GeForce GTX 1070 
Memory: 15978MiB

**Comment by Euroblitz** [comment]
Same issue on native Gentoo Steam

OS: Gentoo Linux x86_64 
Host: B450 AORUS M 
Kernel: 6.1.31-gentoo-dist 
Resolution: 1920x1080 
DE: Plasma 5.27.5 
CPU: AMD Ryzen 5 3600 (12) @ 3.600GHz 
GPU: AMD ATI Radeon RX 6600 XT
Memory: 15906MiB

**Comment by InventorXtreme** [comment]
Same issue here on stable

OS: Arch Linux x86_64
Kernel 6.3.9-arch1-1
CPU: 8-core AMD Ryzen 7 5800X
Memory 32GB
GPU: RTX 3070
DE: X11 BSPWM

**Comment by truckerboy** [comment]
Same issue on Ubuntu 22.04

Steam client version: 1687386907
Operating System: Ubuntu 22.04
Gnome Version: 42.5
Kernel Version: 5.19.0-46-generic #47~22.04.1-Ubuntu x86_64
Graphics Platform: Wayland
Resolution: 3840x2160 (16:9)
CPU: Intel® Core™ i7-10710U CPU @ 1.10GHz × 12
Memory: 64 GiB of RAM
GPU: NVIDIA GeForce RTX 3060

**Comment by FloriansCode** [comment]
Same issue on Manjaro/Gnome

Steam client version: 1687386907 (via Manjaro Repo)
OS: Manjaro Linux x86_64
Kernel: 6.1.31-2-MANJARO
DE: GNOME 44.1
CPU: AMD Ryzen 5 3600 (12) @ 3.600GHz
GPU: AMD Radeon RX 5600 XT
Graphics Platform: Wayland

**Comment by wilkemeyer** [comment]
Same Issue on Ubuntu 23.04 

Steam Clent Version: 1689034492
OS: Ubuntu 23.04 (amd64)
Kernel: 6.4.3 (own config, build with llvm17, full-lto)
DE: KDE Plasma 5.27.6  (active PPA: kubuntu-ppa/backports)
CPU: AMD Ryzen 7 5800X 8-Core Processor
GPU: AMD Radeon RX 6800 XT
Graphics Platform: X11 (also happend on Wayland)

Mesa: Mesa 23.1.3 - kisak-mesa PPA

**Comment by zharvz** [comment]
Same here

**Comment by Sienile** [comment]
Same on a fresh Ubuntu system. Ryzen 7, AMD RX 570

**Comment by sonofevil** [comment]
Has been randomly happening for as long as I've been using Steam on Linux.

Steam Client: 1689814451
Operating System: EndeavourOS 
KDE Plasma Version: 5.27.6
KDE Frameworks Version: 5.107.0
Qt Version: 5.15.10
Kernel Version: 6.1.35-1-lts (64-bit)
Graphics Platform: X11
Processors: 4 × Intel® Core™ i5-4300U CPU @ 1.90GHz
Memory: 7,5 GiB of RAM
Graphics Processor: Mesa Intel® HD Graphics 4400
Manufacturer: LENOVO
Product Name: 20ARS1M901
System Version: ThinkPad T440s

**Comment by rKsanu2MMYvypWePtQWM** [comment]
This is comical at this point - there was the UI beta period, in which this bug existed. I guess a black screen wasn't important enough to fix it before going to stable, even now it's not fixed. I have 3 Linux computers, including the deck and all show this behavior. It just doesn't work.

I'm sure this would've been fixed weeks ago if at least a single person in the Steam client development team tested the Linux build for at least 5 minutes.

**Comment by erpadmin** [comment]
confirming on manjaro kde;  any time screen decides to blank (usually when i navigate back to the client which i size at 3/4 res) i click library and resize the window and steamui starts updating all tabs again

**Comment by allora** [comment]
Same blank screen. For me this is most common when i switch desktop work spaces (i use BSPWM and make heavy use of desktop switching)

Steam Version:  1690583737
Steam Client Build Date:  Fri, Jul 28 11:44 AM UTC -08:00
Steam Web Build Date:  Fri, Jul 28 3:21 PM UTC -08:00
Steam API Version:  SteamClient020

OS: Arch
Kernel: Linux rufus 6.4.7-arch1-2
DE: N/A
WM: bspwm
CPU: AMD Ryzen 9 5950X 16-Core Processor
GPU: AMD Radeon RX 6700 XT
Graphics Platform: X11
Memory: 64 GB

**Comment by gwstorm** [comment]
Steam Version:  1690583737
Steam Client Build Date:  Fri, Jul 28 11:44 AM UTC -08:00
Steam Web Build Date:  Fri, Jul 28 3:21 PM UTC -08:00
Steam API Version:  SteamClient020

OS: Arch
Kernel: Linux 6.4.8-arch1-1
DE: N/A
WM: bspwm
CPU: AMD Ryzen 9 5950X 16-Core Processor
GPU: AMD Radeon RX 7900XTX
Graphics Platform: X11
Memory: 32 GB

I've been dealing with this since the update and while this is a bit of a first-world problem, it's insanely ******* annoying. I have a 49" super ultra wide monitor with a dark theme throughout and steam flash bangs me with a bright white every time I cross the workspace with steam on it.

**Comment by kubaa2002** [comment]
I have been experiencing the same issue

Steam Version:  1690583737
Steam Client Build Date:  Fri, Jul 28 20:44 UTC -08:00
Steam Web Build Date:  Sat, Jul 29 00:21 UTC -08:00
Steam API Version:  SteamClient020

OS: Nobara 
kernel: 6.3.12
DE: gnome
CPU: ryzen 7 7700
GPU: Radeon RX 7600

**Comment by illusioon** [comment]
Same problem, Blank UI when minimize or maximize.
Operating System Version:
"Void Linux" (64 bit)
Kernel Name: Linux
Kernel Version: 6.3.13_1
X Server Vendor: The X.Org Foundation
X Server Release: 12101008
X Window Manager: Xfwm4
Steam Runtime Version: steam-runtime_0.20230606.51628

**Comment by TomCoursow** [comment]
Same here... Have this issues now for multiple years... This is extremely annoying...

**Comment by aravlo** [comment]
Same here, minimizing the Steam Client window on the Activity or Store Page (Browser Pages I suppose) and then bringing it back into the foreground makes the Window only display a black box where the content should be. Refreshing the page, or switching between Store and Activity Tabs does not change this. Only switching to the Library and then back to either Store/Activity makes the content appear.

OS: Manjaro Uranos 23.0.0
Kernel: 6.1.44-Manjaro #1
GPU: AMD Navi 21 [Radeon RX 6800/6800 XT / 6900 XT]
Driver: amdgpu (Mesa 23.1.5)
Display: X11 - 1.21.1.8
DE: Mate 1.26
WM: Metacity
Distribution Steam Package: 1.0.0.78-2

Steam Version:  1690583737
Steam Client Build Date:  Fri, Jul 28 20:44 UTC -08:00
Steam Web Build Date:  Sat, Jul 29 00:21 UTC -08:00
Steam API Version:  SteamClient020

**Comment by rhoot** [comment]
Same here. Been happening ever since the switch to the new UI. It starts out fine but as soon as the window is resized, it goes black and stays that way until I go to the Library and back. Happens in both KDE and sway.

"EndeavourOS Linux" (64 bit)
Kernel Name: Linux
Kernel Version: 6.4.12-zen1-1-zen
X Server Vendor: The X.Org Foundation
X Server Release: 12302000
X Window Manager: KWin
Steam Runtime Version: steam-runtime_0.20230606.51628

**Comment by peppot** [comment]
Same on stable.

**Comment by Shensen1** [comment]
can confirm the same bug in NixOS with KDE

**Comment by nikhil-prabhu** [comment]
Same here. However, for me, going to the library and back to the store page doesn't always work. I have to go to a game in my library, go to its store page, and then navigate back to the store home page.

Operating System: openSUSE Tumbleweed 20231215
KDE Plasma Version: 5.27.10
KDE Frameworks Version: 5.112.0
Qt Version: 5.15.11
Kernel Version: 6.6.6-1-default (64-bit)
Graphics Platform: Wayland
Processors: 8 × AMD Ryzen 5 3400G with Radeon Vega Graphics
Memory: 31.3 GiB of RAM
Graphics Processor: AMD Radeon RX 6700 XT

FWIW, I switched to Wayland just around a month ago, and didn't really face this issue with X11 (At least not the black/blank store page. The overall UI responsiveness was still quite bad, though).

**Comment by LukeDearden** [comment]
Hi,

I think I found a weird workaround for this. 

Don't maximize Steam. Keep it in a Window. 

For me the Store etc work fine and in game browser/profile viewing is fine. 

Operating System Version:
Fedora Linux 39 (Workstation Edition) (64 bit)
Kernel Name: Linux
Kernel Version: 6.6.8-200.fc39.x86_64
X Server Vendor: The X.Org Foundation
X Server Release: 12302003
X Window Manager: GNOME Shell
Steam Runtime Version: steam-runtime_0.20231127.68515

Steam Beta Branch:  Steam Beta Update
Steam Version:  1703100617
Steam Client Build Date:  Wed, Dec 20 21:06 UTC -08:00
Steam Web Build Date:  Wed, Dec 20 20:14 UTC -08:00
Steam API Version:  SteamClient021

**Comment by Root-Core** [comment]
Quote @deanrih:
> Unresponsiveness

This seems to be fixed, I can not reproduce the issue currently.
CC @kisak-valve

> Blank part the UI

This issue is handled in #9598 and still not fixed.
As it these two issues aren't directly related, it might be better handled there.

---------------

Quote @LukeDearden:
> I think I found a weird workaround for this.
> 
> Don't maximize Steam. Keep it in a Window.

Yeah, that's already consensus. It also works, if you only use the maximize button on Steam's window.
It does not work, if you maximize it by dragging it to the upper border of your screen or maximize it externally via. your desktop.

You might want to take a look at #9598.

**Comment by dsalt** [comment]
> It also works if you only use the maximize button on Steam's window.

Which is stupidly broken because that button doesn't allow for along one axis only, so I normally toggle maximisation via the window manager. Though usually I only have the window maximised along one axis (usually vertical), which isn't enough to trigger the bug.

> It does not work, if you maximize it by dragging it to the upper border of your screen

Well, yes. Any method involving the window manager and maximisation along both axes is going to trigger the bug.

It only matters that the window ends up maximised along both axes via external (to Steam) methods – timing doesn't matter – which is why I wonder whether the bug is to do with Steam mishandling two maximisation events without an unmaximisation in between. And as this is what (I find) window managers normally do…

**Comment by Root-Core** [comment]
Yeah, every external maximization is problematic here. I guess they fixed the event of their own button (which had the same bug), but they did not fix the external events. But we can only speculate.. it is just really, really annoying.

-----------

Btw.. I wanted to list all (potential) duplicates of the black browser issue, but there are too many.
Here are some: #9375 #9598 #9386 #9598 #10229

**Comment by skuzad25** [comment]
Having issues with Steam UI, it's a blank screen on my Ubuntu 23.04 virtual private server. Friends list appears, Steam store page can be interacted with if you click randomly but it's too difficult to navigate like that.

**Comment by skuzad25** [comment]
Running Steam big picture mode worked for some reason.

**Comment by Ricaz** [comment]
This happened to me after enabling GPU acceleration. In Settings -> Interface, untick this:

- [ ] Enable GPU accelerated rendering in web views (requires restart)

I was able to find the settings menu even with the black screen (drop-downs still render when hovered).

**Comment by Terseus** [comment]
> This happened to me after enabling GPU acceleration. In Settings -> Interface, untick this:
> 
>     * [ ]  Enable GPU accelerated rendering in web views (requires restart)
> 
> 
> I was able to find the settings menu even with the black screen (drop-downs still render when hovered).

If you right click in the Steam systray icon -> Settings it will show the settings window, from there I've deactivated the GPU accelerated rendering in web views and it worked again.

Thanks ^_^

**Comment by rKsanu2MMYvypWePtQWM** [comment]
> > This happened to me after enabling GPU acceleration. In Settings -> Interface, untick this:
> > ```
> > * [ ]  Enable GPU accelerated rendering in web views (requires restart)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I was able to find the settings menu even with the black screen (drop-downs still render when hovered).
> 
> If you right click in the Steam systray icon -> Settings it will show the settings window, from there I've deactivated the GPU accelerated rendering in web views and it worked again.
> 
> Thanks ^_^

Did not work for me

**Comment by eNTi** [comment]
Issue just came back... had this WAAAY back and now it reappeared with:

Steam Beta Branch:  Stable Client
Steam Version:  1709168962
Steam Client Build Date:  Thu, Feb 29 01:39 UTC -08:00
Steam Web Build Date:  Mon, Feb 26 21:21 UTC -08:00
Steam API Version:  SteamClient021

**Comment by InsaneJ** [comment]
I'm getting the same issue.

Steam Beta Branch:  Steam Beta Update
Steam Version:  1709757778
Steam Client Build Date:  Wed, Mar 6 21:28 UTC -08:00
Steam Web Build Date:  Tue, Mar 5 22:57 UTC -08:00
Steam API Version:  SteamClient021

Running Manjaro, currently up to date.
glxinfo -B shows the following:

```
name of display: :1
display: :1  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: AMD (0x1002)
    Device: AMD Radeon RX 7900 XTX (radeonsi, navi31, LLVM 16.0.6, DRM 3.57, 6.7.7-1-MANJARO) (0x744c)
    Version: 24.0.2
    Accelerated: yes
    Video memory: 24576MB
    Unified memory: no
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
Memory info (GL_ATI_meminfo):
    VBO free memory - total: 23080 MB, largest block: 23080 MB
    VBO free aux. memory - total: 15347 MB, largest block: 15347 MB
    Texture free memory - total: 23080 MB, largest block: 23080 MB
    Texture free aux. memory - total: 15347 MB, largest block: 15347 MB
    Renderbuffer free memory - total: 23080 MB, largest block: 23080 MB
    Renderbuffer free aux. memory - total: 15347 MB, largest block: 15347 MB
Memory info (GL_NVX_gpu_memory_info):
    Dedicated video memory: 24576 MB
    Total available memory: 40194 MB
    Currently available dedicated video memory: 23080 MB
OpenGL vendor string: AMD
OpenGL renderer string: AMD Radeon RX 7900 XTX (radeonsi, navi31, LLVM 16.0.6, DRM 3.57, 6.7.7-1-MANJARO)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 24.0.2-manjaro1.1
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6 (Compatibility Profile) Mesa 24.0.2-manjaro1.1
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 24.0.2-manjaro1.1
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

**Comment by Root-Core** [comment]
We have a regression here, these tabs black out again.. even if you use the controls provided by Steam or externally via KWin (maximize/minimize via taskbar).

**Comment by mhanuszh** [comment]
> We have a regression here, these tabs black out again.. even if you use the controls provided by Steam or externally via KWin (maximize/minimize via taskbar).

Regression? For me it never worked since the new UI update.

**Comment by Root-Core** [comment]
> Regression? For me it never worked since the new UI update.

How did you resize the window?
It was working if you used the controls in the upper right corner of the Steam window, not if you moved it to the upper border of the desktop or maximized it externally.

These controls were working after some fix in September 2023, they aren't anymore. Have a look at #9598

**Comment by mhanuszh** [comment]
> > Regression? For me it never worked since the new UI update.
> 
> How did you resize the window? It was working if you used the controls in the upper right corner of the Steam window, not if you moved it to the upper border of the desktop or maximized it externally.
> 
> These controls were working after some fix in September 2023, they aren't anymore. Have a look at #9598

I always maxmize with either dragging the window to the top or by Meta+UpArrow, so I didnt even knew that the buttons were also busted, then fixed, and now broken again.

It looks like Valve is trying something out with the windowing code, given the recent CEF update. Or, at least I'm hoping that it's only their QA being sloppy and not the spaghetti monster coming to life.

Anyway, I'm hoping that someday I can maximize the store without Steam blacking out.

**Comment by Root-Core** [comment]
Another issue, same problem: If I minimize a maximized Steam window and restore it, it snaps into a non-maximized window. Can somebody confirm this behavior?

----------

@mhanuszh  Yeah, I usually do the same. Not with Steam though.. we can just wait and hope.

**Comment by radionaut** [comment]
> Another issue, same problem: If I minimize a maximized Steam window and restore it, it snaps into a non-maximized window. Can somebody confirm this behavior?
confirming this. fedora 39, KDE wayland, steam stable client (dnf)

**Comment by radionaut** [comment]
> Another issue, same problem: If I minimize a maximized Steam window and restore it, it snaps into a non-maximized window. Can somebody confirm this behavior?

confirming this. fedora 39, KDE wayland, steam stable client (dnf)

**Comment by GnomeBeans** [comment]
Same issue as everyone else. It does not want to work maximized. I can resize the screen and it will play video fullscreen without turning black. I tried without GPU acceleration like suggested and this instead causes the screen to flicker. It also resizes itself like other people have described when you've maximize and minimize. 

I use multi monitor arch with latest update, kernel and KDE 6. AMD 6800 XT radv driver. I am using repo steam not flatpak. I wonder if those people are having the same problems or if it'll fix this behavior?

**Comment by Ismas** [comment]
Same issue here. Ubuntu 22.04 LTS. Regression to blank/black screen, all screens except settings. Buttons worked but cannot be seen. Once even hang the Nvidia driver.

SOLUTION: Settings -> Interface -> Enable smooth scrolling in web views -> DISABLE.

It worked for me. You mileage yada yada yada.

**Comment by mhanuszh** [comment]
Still happening `with / without` `smooth scrolling / GPU acceleration / hw video decoding` by maximizing with `mouse / shortcut / window control buttons`.
 
<details><summary>Debian 12, Nvidia 525.147.05, Gnome 43.9, X11, Kernel 6.1.0-17-amd64</summary>

Steam:
Steam Beta Branch:  Steam Beta Update  
Steam Version:  1710786209  
Steam Client Build Date:  Mon, Mar 18 19:05 UTC -08:00  
Steam Web Build Date:  Mon, Mar 18 19:09 UTC -08:00  
Steam API Version:  SteamClient021  

Computer Information:
Manufacturer: Dell Inc.
Model: 06D7TR
Form Factor: Desktop
No Touch Input Detected

Processor Information:
CPU Vendor: GenuineIntel
CPU Brand: Intel(R) Core(TM) i5-2500 CPU @ 3.30GHz
CPU Family: 0x6
CPU Model: 0x2a
CPU Stepping: 0x7
CPU Type: 0x0
Speed: 3700 MHz
4 logical processors
4 physical processors
Hyper-threading: Unsupported
FCMOV: Supported
SSE2: Supported
SSE3: Supported
SSSE3: Supported
SSE4a: Unsupported
SSE41: Supported
SSE42: Supported
AES: Supported
AVX: Supported
AVX2: Unsupported
AVX512F: Unsupported
AVX512PF: Unsupported
AVX512ER: Unsupported
AVX512CD: Unsupported
AVX512VNNI: Unsupported
SHA: Unsupported
CMPXCHG16B: Supported
LAHF/SAHF: Supported
PrefetchW: Unsupported

Operating System Version:
Debian GNU/Linux 12 (bookworm) (64 bit)
Kernel Name: Linux
Kernel Version: 6.1.0-17-amd64
X Server Vendor: The X.Org Foundation
X Server Release: 12101007
X Window Manager: GNOME Shell
Steam Runtime Version: steam-runtime_0.20240304.79797

Video Card:
Driver: NVIDIA Corporation NVIDIA GeForce GTX 1050 Ti/PCIe/SSE2
Driver Version: 4.6.0 NVIDIA 525.147.05
OpenGL Version: 4.6
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 60 Hz
VendorID: 0x10de
DeviceID: 0x1c82
Revision Not Detected
Number of Monitors: 1
Number of Logical Video Cards: 1
Primary Display Resolution: 1920 x 1080
Desktop Resolution: 1920 x 1080
Primary Display Size: 18.74" x 10.55" (21.50" diag), 47.6cm x 26.8cm (54.6cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 4096 MB
Supported MSAA Modes: 2x 4x 8x 16x

Sound card:
Audio device: Realtek ALC269VB

Memory:
RAM: 15947 Mb
VR Hardware:
VR Headset: None detected
Miscellaneous:
UI Language: English
LANG: en_GB.UTF-8
Total Hard Disk Space Available: 408278 MB
Largest Free Hard Disk Block: 32091 MB

Storage:
Number of SSDs: 3
SSD sizes: 500G,500G,500G
Number of HDDs: 0
Number of removable drives: 0
</details>

**Comment by bphenriques** [comment]
> Same issue here. Ubuntu 22.04 LTS. Nvidia 4070TI, driver 535.161.07. Regression to blank screen, all screens black except settings. Buttons work but cannot be seen. Mine broken maximized and not maximized. Once even hanged the Nvidia driver.
> 
> SOLUTION: Settings -> Interface -> Enable smooth scrolling in web views -> DISABLE.
> 
> It worked for me. You mileage yada yada yada.

This worked for me. The contextual menu appeared and I was able to enter the settings. You have to click and move the mouse around the top left corner until you see a Settings menu item

**Comment by dsalt** [comment]
> Same issue here. Ubuntu 22.04 LTS. Nvidia 4070TI, driver 535.161.07. Regression to blank screen, all screens black except settings. Buttons work but cannot be seen. Mine broken maximized and not maximized. Once even hanged the Nvidia driver.

More than just the browser component being blank makes that a separate issue.

This bug report is specifically about the browser component (and **not** the rest of the UI) becoming blank when fully maximised or after workspace switching.

**Comment by fornellas** [comment]
Steam opens like this:

![Screenshot at 2024-04-07 14-43-04](https://github.com/ValveSoftware/steam-for-linux/assets/1386232/a3a862e5-75d6-484e-a3b2-7884bb3c0b6b)

It is responsive, if I hover over the menu it does show the items.

Ubuntu 22.04.4 LTS, nvidia-driver-550.

This issue used to happen only after suspend / resume, now it happens all the time: **Steam is 100% unusable as a result**.

**Comment by kisak-valve** [comment]
Hello @fornellas, you're showing #10543. If that's true, then you can try running Steam from a terminal with `-system-composer` or `-cef-disable-gpu` as a workaround.

**Comment by BloodyIron** [comment]
I get this black screen bs all the time and it's infuriating. Cannot find a solution for it.

**Comment by dsalt** [comment]
Blank browser component:

* **Window maximisation:** appears to be fixed, as of 1714160074.
* **Switching workspaces:** on switching back to the workspace with the Steam window, the content is briefly shown before it's blanked.

**Comment by Root-Core** [comment]
I can confirm window maximization is finally working now, thanks!

Switching workspaces works for me too on KDE Plasma 6.0.4 and Steam 1714160074

**Comment by BloodyIron** [comment]
Version I'm using:

Steam Beta Branch:  Steam Beta Update
Steam Version:  1714160074
Steam Client Build Date:  Fri, Apr 26 1:10 PM UTC -08:00
Steam Web Build Date:  Fri, Apr 26 1:13 PM UTC -08:00
Steam API Version:  SteamClient021

I just now tried to reproduce the black-screen problem... and so far I cannot seem to reproduce it like I was able to before! YAY!!!! Hopefully this is a reliable fix, and thanks to VALVe for maybe fixing it! :)))

**Comment by rKsanu2MMYvypWePtQWM** [comment]
Seems to work for me too, though I did switch to wayland with KDE6 so I can't reliably state what exactly fixed it.

**Comment by zsoki** [comment]
I can also confirm, that in Steam beta version 1714160074, the black/blank screen issue is fixed.
I'm using OpenSUSE Tumbleweed, KDE Plasma 6.0.4 Wayland, Kernel version 6.8.7-1-default (64-bit), AMD Radeon RX 7800 XT

**Comment by InsaneJ** [comment]
Now running version 1714520927 and the problem seems to have been resolved for me as well. Thanks!

**Comment by deanrih** [comment]
Can confirm that version `1714520927` (haven't checked at which version this is fixed) no longer has this issue, and it's quite responsive as well, Arch - X11 - NVIDIA

Might close this issue if there are no further discussions.

**Comment by MartinX3** [comment]
@deanrih 
I think it should only be closed if it works on the standard update branch.
They may break it while merging the beta branch.

**Comment by deanrih** [comment]
> @deanrih 
> I think it should only be closed if it works on the standard update branch.
> They may break it while merging the beta branch.

oh right, I forgot I was also in the beta branch, good catch

**Comment by dsalt** [comment]
> Might close this issue if there are no further discussions.

As I mentioned above, it's only partly fixed.

**Comment by Root-Core** [comment]
> > Might close this issue if there are no further discussions.
> 
> As I mentioned above, it's only partly fixed.

I would argue that this is a different bug and it seems to be Xfce specific.
Better open a new issue, as the here described bug is fixed in the beta.

**Comment by DanielHuelsmann** [comment]
After upgrading to the latest **stable** version of Steam **(version 1714854927),** this bug is fixed. I can maximise, minimise and resize the Steam window; without the different sections (shop, community, _profile_) turning black.

It works very well now.

**Comment by DanielHuelsmann** [comment]
In the latest Steam Client Update (May 7th), it is mentioned in the Linux section:

https://store.steampowered.com/news/app/593110/view/4182231197207095612

**Comment by rkfg** [comment]
Still broken on:
Steam Beta Branch:  Stable Client
Steam Version:  1715635533
Steam Client Build Date:  Tue, May 14 00:01 UTC +03:00
Steam Web Build Date:  Mon, May 13 21:06 UTC +03:00
Steam API Version:  SteamClient021

The beta branch is at the same version, I checked. The main window is pitch black but I can interact with menus and open settings etc. My NVIDIA driver version is 550.78.

**Comment by MartinX3** [comment]
Yes, it's still broken.

**Comment by ironashram** [comment]
works perfectly fine for me with 1715891371

**Comment by jmfergeau** [comment]
I just got the same problem today. Pitch black window while settings window works. Disabling smooth scrolling didn't fix it.

**Comment by clemzarch** [comment]
I've been getting this bug on Manjaro, and on Steam Deck in desktop mode (since April 2022).
Some days ago I factory-resetted an OLED Deck (for unrelated audio bugs), and it still happens on the Stable branch, with HW accel and smooth scrolling enabled.

**Comment by TomCoursow** [comment]
Now it's completely broken for me... Not a single Steam Client Web Page (Profile, Community, Shop and so on) works anymore... Only black screens.
Switching between library and shop does also not help anymore. Min/Maxing Window does also not help

`
Steam-Betabranch:  Steam Beta Update
Steam-Version:  1721420033
Steam-Client: Build-Datum:  Fr., 19. Juli 21:31 UTC -08:00
Steam: Webbuild-Datum:  Fr., 19. Juli 21:30 UTC -08:00
Steam-API-Version:  SteamClient021
`

**Comment by MartinX3** [comment]
I think the main problem is the massively outdated chromium version integrated in steam.

**Comment by peppot** [comment]
Happening here today. Restarted to update Steam, no change. Ubuntu 22.04

**Comment by rkfg** [comment]
It's possible to have webview GPU acceleration enabled and still use Steam. Run a compositing manager such as compton/picom (a modern fork) and then run Steam with GPU acceleration enabled. Should do the trick.

**Comment by readtedium** [comment]
Having this issue as well. Turning off GPU acceleration on web views fixes it, but kills the performance of Big Picture mode.

**Comment by dsalt** [comment]
I think that it's safe to close this as the maximisation blanking bug has been fixed for a while now.

**Comment by saffroy** [comment]
It isn't fixed for me: switching desktops (among other things) always yields black web views. 

Environment: Steam beta update (1723685248), GPU disabled on web views, Xfce 4.18 on Debian 12.

I don't see how this would be a different bug, though I don't really care, I'd just need to know which issue to track instead.

**Comment by readtedium** [comment]
@dsalt The subject is “[Steam UI] Unresponsive and black/blank screen on the new UI,” which is the issue I am (and seemingly a lot of other people are) having.

This should have a different subject line if it’s something else.

**Comment by dsalt** [comment]
My reading of the initial comment:

1. Bug concerning maximisation of Steam windows with browser components.
2. Bug involving initial unresponsiveness of Steam's dedicated browser window.

Two bugs, possibly related.

No mention of workspace-switching (desktop-switching). I've now reported this one as #11198.

Some nvidia users reported in the comments a third bug in which the *whole* of the Steam window, not merely the browser content area, became black. (Example: https://github.com/ValveSoftware/steam-for-linux/issues/9386#issuecomment-2041476706.) I know not the cause of this one, but I suspect nvidia's drivers.

**Comment by theNoof** [comment]
i have a similar issue:
when i start steam, i get a black window. i can open the dropdown menu with right click, but the window stays black.

some system stats:
- nixos 24.11
- intel graphics/cpu
- wayland compositor (niri wm)
- using xwayland-sattelite for x11 compatibility

is there a fix i missed?

**Comment by rkfg** [comment]
I run it with `-system-composer` now (single dash, yes), have you tried this option?

**Comment by theNoof** [comment]
yes, doing that. changes nothing.

**Comment by Thespikedballofdoom** [comment]
only -system-composer fixed it for me.

**Comment by retblast** [comment]
This happens to me on Intel + Gnome + Wayland. Desktop switching triggers it, disabling gpu rendering or -system-composer do not fix this.

**Comment by 3Lord3** [comment]
> This happens to me on Intel + Gnome + Wayland. Desktop switching triggers it, disabling gpu rendering or -system-composer do not fix this.

It's exactly the same for me
